composer/loggers/neptune_logger.py

```diff
@@ -232,7 +240,11 @@ def upload_file(
             return
 
         del state  # unused
-        self.base_handler[remote_file_name].upload(str(file_path))
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_file_path = os.path.join(temp_dir, str(file_path.name))
+            shutil.copy2(file_path, temp_dir)
+            self.base_handler[remote_file_name].upload(temp_file_path)
```

Comment:
As per the `upload_file` docstrings:
```
 After this method returns, training can resume, and the contents of ``file_path`` may change (or be may
  deleted). Thus, if processing the file in the background (as is recommended), it is necessary to first
  copy the file to a temporary directory. Otherwise, the original file may no longer exist, or the logged
  file can be corrupted (e.g., if the logger destination is reading from file while the training loop
  is writing to it).
```